examples/nidigital_spi/teststand_fixture.py

```diff
@@ -0,0 +1,120 @@
+    """Update registered pin map contents.
+
+    Create and register a pin map if a pin map resource for the specified pin map id is not found.
+
+    Args:
+        pin_map_id (str): The resource id of the pin map to register as a pin map resource. By
+            convention, the pin_map_id is the .pinmap file path.
+
+    """
+    with GrpcChannelPoolHelper() as grpc_channel_pool:
+        pin_map_client = PinMapClient(grpc_channel=grpc_channel_pool.pin_map_channel)
+        pin_map_client.update_pin_map(pin_map_id)
+
+
+def create_nidigital_sessions(
+    pin_map_id: str,
+    specifications_file_paths: Iterable[str],
+    levels_file_paths: Iterable[str],
+    timing_file_paths: Iterable[str],
+    pattern_file_paths: Iterable[str],
+) -> None:
+    """Create and register all NI-Digital sessions."""
+    with GrpcChannelPoolHelper() as grpc_channel_pool:
+        session_management_client = nims.session_management.Client(
+            grpc_channel=grpc_channel_pool.session_management_channel
+        )
+
+        pin_map_context = nims.session_management.PinMapContext(pin_map_id=pin_map_id, sites=None)
+        with session_management_client.reserve_sessions(
+            context=pin_map_context,
+            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DIGITAL_PATTERN,
+            # This code module sets up the sessions, so error immediately if they are in use.
+            timeout=0,
+        ) as reservation:
+            for session_info in reservation.session_info:
+                grpc_options = nidigital.GrpcSessionOptions(
+                    grpc_channel_pool.get_grpc_device_channel(
+                        nidigital.GRPC_SERVICE_INTERFACE_NAME
+                    ),
+                    session_name=session_info.session_name,
+                    initialization_behavior=nidigital.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
+                )
+
+                session = nidigital.Session(
+                    resource_name=session_info.resource_name, grpc_options=grpc_options
+                )
+                # Preload the pin map, specifications, levels, timing, and pattern files.
+                session.load_pin_map(pin_map_id)
+                session.load_specifications_levels_and_timing(
+                    [
+                        str(_resolve_relative_path(service_directory, path))
+                        for path in specifications_file_paths
+                    ],
+                    [
+                        str(_resolve_relative_path(service_directory, path))
+                        for path in levels_file_paths
+                    ],
+                    [
+                        str(_resolve_relative_path(service_directory, path))
+                        for path in timing_file_paths
+                    ],
+                )
+                for path in pattern_file_paths:
+                    session.load_pattern(str(_resolve_relative_path(service_directory, path)))
```

Comment:
NI-Digital returns an error if I try to load the specifications more than once without unloading, which seems like a pretty clear hint that this should only be done once per TestStand execution.

```
nidigital.errors.DriverError: -1074118560: An error occurred while loading a specifications file.

Device: DigitalPattern1
spml_kUsageTypeErrorMessage: Failed to load specifications file D:\dev\measurementlink-python\examples\nidigital_spi\Specifications.specs. Found duplicate symbol 'dc.vcc'
```

In TestStand, I'm currently specifying the filenames as file globals:
![image](https://user-images.githubusercontent.com/17914399/232659057-c8b6fbe1-50cf-4780-ae3f-153e5e19a433.png)

Cc: @jonathanmendez @rfriedma

Reply:
I haven't gotten any feedback suggesting that we do otherwise, so I'm marking this as resolved.